Add ARIA "img" role for temporary placeholder

ssorallen opened the pull request:
Because the placeholder is intended as a img replacement while loading
is happening, it should be treated as an img by screen readers.

Fixes #12.

Commit:
Add ARIA "img" role for temporary placeholder

Because the placeholder is intended as a img replacement while loading
is happening, it should be treated as an img by screen readers.

Fixes #12.

ssorallen commented:
Actually since that is in the shadow DOM this might not be sufficient. Should `core-image` be setting `role="img"` on itself at all times?

cdata commented:
This raises an interesting question for me. What is the relationship between elements in a shadow and a screen reader? Does the screen reader skip over them in favor of whatever it can find in the light-most side of the DOM? Or will it dive deeply into shadow as though it were normal DOM?

cdata commented:
Very interesting: https://www.youtube.com/watch?v=2UfabypzfGU (JSBin @ http://jsbin.com/votota).

This suggests that it isn't especially helpful to define ARIA roles on shadow children. It also raises a lot of much larger questions for me about a11y in a Shadow DOM world.

ssorallen commented:
[According to @alice](https://github.com/Polymer/core-image/issues/12#issuecomment-75897427), some screen readers do look at the shadow DOM's composed tree. @cdata, that doesn't exactly fit your demo though. I'm unsure what the appropriate way forward is?

cdata commented:
Well, my demo was using ChromeVox, so it sounds like I need to get a "real" screenreader ;)

Let's put `role="img"` on the host and `role="none"` on the `<img>`. There is probably more subtle logic we could use to decide whether we need `role="img"` on the host or not, but let's save it for 0.8.

Commit:
Apply ARIA role to entire element for consistency

Rather than add/remove the `role` attribute from descendant elements,
apply it once to the element itself and disable the descendant `<img>`
from acting like an image. This provides a consistent API for the
element, and screenreaders don't need to inspect the shadow DOM.

Commit:
Use single quotes to match file style

cdata commented:
:+1:

alice commented:
That's right: ChromeVox right now doesn't support Shadow DOM. Work is underway to fix that :)
